File: 392_is_subsequence/main.py

class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        sequence = ''
        start_t, start_s = 0, 0
        for idx_s in range(start_s, len(s)):
            for idx_t in range(start_t, len(t)):
                if t[idx_t] == s[idx_s]:
                    start_s += 1
                    start_t = idx_t
                    sequence += s[idx_s]
                    break
            start_t += 1
        return s == sequence
        
solution = Solution()
print(solution.isSubsequence(s = "aaaaaa", t = "bbaaaa"))

File: 392_is_subsequence/main.go
package main

import "fmt"

func isSubsequence(s string, t string) bool {
	a := ""
	idx_t := 0
	idx_s := 0
	for j := idx_s; j < len(s); j++ {
		for i := idx_t; i < len(t); i++ {
			if s[j] == t[i] {
				idx_s += 1
				idx_t = i
				a += string(s[j])
				break
			}
		}
		idx_t += 1
	}
	return a == s
}

func main() {
	s := "abc"
	t := "ahbgdc"
	boolean := isSubsequence(s, t)
	fmt.Println(boolean)
}